### ISSUE TITLE
handle case when CL_R channel order is not supported  

### ISSUE THREAD

**Issue by maweigert** [ClearVolume/ClearVolume#41]
On some older GPU CL_R is not in the supported image channel orders and an excpetion is raised when creating the volumetric texture - we should use RGBA then.

**Comment by skalarproduktraum** [comment]
Do you know what GPUs are affected by this? I remember only seeing this on Intel Integrated Graphics 3000, which only has 384MB of RAM and might therefore be not too useful anyway.

**Comment by maweigert** [comment]
Yes, I remember it to be an integrated Intel HD 3000 - which early 2011 MacBook Pros without dedicated GPU seem to be working on. As for the 384MB, afaik RAM size is not fixed anyway for integrated cards as they take it from the main memory and that share is adjustable. But you're right, only few users of CV with such a setup would enjoy the render speed even with a supported image format. Any case, I'd rate it low priority.

**Comment by skalarproduktraum** [comment]
I'd vote for closing the issue. If we have to use RGBA, this would increase the amount of memory needed by a factor of four. Considering these machines are very memory constrained anyway, it would mean the approximately useful volume size would be something ~256MiB. Alternatively, we could also use CL_LUMINANCE, which was supported on the old GPUs if I'm not mistaken. Merit unclear, nevertheless.

**Comment by skalarproduktraum** [comment]
On a related note, did you benchmark performance image buffers vs. regular buffers for volume rendering by chance? Using regular buffers would also solve this issue, but would probably introduce indexing overhead.

**Comment by maweigert** [comment]
1) As for closing: I still would rather see CV rendering slow than just throw an exception in this case. In the meantime I've downgraded the issue to "enhancement" rather than "bug" to not let anyone trying to fix the latter waste time on that non important one ;)  
2) Buffers vs textures: I did some benchmarking in the past indicating that for the random orientation memory traversal we use for the max project rendering the texture cache is indeed better suited and faster than the possible uncoalesced traversal for buffers (e.g. when rendering along the z axis). Plus interpolation comes in for free what made the implementation looking cleaner.

**Comment by skalarproduktraum** [comment]
1) :+1: 
2) thanks for the info, so let's keep it the way it is.
